src/modules/ChannelList/context/hooks/useFetchChannelList.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback } from 'react';
+import { GroupChannel, GroupChannelListQuery } from '@sendbird/chat/groupChannel';
+
+import { Nullable } from '../../../../types';
+import { CustomUseReducerDispatcher, Logger } from '../../../../lib/SendbirdState';
+import { MarkAsDeliveredSchedulerType } from '../../../../lib/hooks/useMarkAsDeliveredScheduler';
+import * as channelListActions from '../../dux/actionTypes';
+import { DELIVERY_RECEIPT } from '../../../../utils/consts';
+
+interface DynamicProps {
+  channelSource: Nullable<GroupChannelListQuery>;
+  premiumFeatureList: string[];
+  disableMarkAsDelivered: boolean;
+}
+interface StaticProps {
+  channelListDispatcher: CustomUseReducerDispatcher;
+  logger: Logger;
+  markAsDeliveredScheduler: MarkAsDeliveredSchedulerType;
+}
+
+export const useFetchChannelList = ({
+  channelSource,
+  premiumFeatureList,
+  disableMarkAsDelivered,
+}: DynamicProps, {
+  channelListDispatcher,
+  logger,
+  markAsDeliveredScheduler,
+}: StaticProps) => {
+  return useCallback(async () => {
+    const setMarkAsDelivered = premiumFeatureList.find((feature) => feature === DELIVERY_RECEIPT);
```

Comment:
ideally this shoudlnt be calculated isndie this hook, but we can let it go for now 😉